src/couch_replicator_scheduler.erl

```diff
@@ -320,40 +345,44 @@ job_by_id(Id) ->
     end.
 
 
--spec reschedule(MaxJobs :: non_neg_integer(), MaxChurn :: non_neg_integer()) -> ok.
-reschedule(MaxJobs, MaxChurn)
-  when is_integer(MaxJobs), MaxJobs >= 0, is_integer(MaxChurn), MaxChurn > 0 ->
+-spec reschedule(#state{}) -> ok.
+reschedule(State) ->
     Running = running_job_count(),
     Pending = pending_job_count(),
-    stop_excess_jobs(MaxJobs, Running),
-    start_pending_jobs(MaxJobs, Running, Pending),
-    rotate_jobs(MaxJobs, MaxChurn, Running, Pending).
-
+    stop_excess_jobs(State, Running),
+    start_pending_jobs(State, Running, Pending),
+    rotate_jobs(State, Running, Pending).
 
-stop_excess_jobs(Max, Running) when Running > Max ->
-    stop_jobs(Running - Max);
 
-stop_excess_jobs(_, _) ->
-    ok.
-
-
-start_pending_jobs(Max) ->
-    start_pending_jobs(Max, running_job_count(), pending_job_count()).
+stop_excess_jobs(State, Running) ->
+    #state{max_jobs=MaxJobs, max_history=MaxHistory} = State,
+    if Running > MaxJobs ->
+        stop_jobs(Running - MaxJobs, MaxHistory);
+    true ->
+        ok
+    end.
 
 
-start_pending_jobs(Max, Running, Pending) when Running < Max, Pending > 0 ->
-    start_jobs(Max - Running);
+start_pending_jobs(State) ->
+    start_pending_jobs(State, running_job_count(), pending_job_count()).
 
-start_pending_jobs(_, _, _) ->
-    ok.
 
-rotate_jobs(MaxJobs, MaxChurn, Running, Pending) when Running == MaxJobs, Pending > 0 ->
-    stop_jobs(min([Pending, Running, MaxChurn])),
-    start_jobs(min([Pending, Running, MaxChurn]));
-
-rotate_jobs(_, _, _, _) ->
-    ok.
+start_pending_jobs(State, Running, Pending) ->
+    #state{max_jobs=MaxJobs, max_history=MaxHistory} = State,
+    if Running < MaxJobs, Pending > 0 ->
+        start_jobs(MaxJobs - Running, MaxHistory);
+    true ->
+        ok
+    end.
 
+rotate_jobs(State, Running, Pending) ->
+    #state{max_jobs=MaxJobs, max_history=MaxHistory, max_churn=MaxChurn} = State,
+    if Running == MaxJobs, Pending > 0 ->
```

Comment:
extract MaxJobs in the function head so the guard clause can work, and extract the others in that `= State` line.